zerver/forms.py

```diff
@@ -50,6 +52,13 @@ class RegistrationForm(forms.Form):
     # actually required for a realm
     password = forms.CharField(widget=forms.PasswordInput, max_length=100,
                                required=False)
+    realm_name = forms.CharField(max_length=100, required=False)
+    if settings.REALMS_HAVE_SUBDOMAINS:
+        realm_url = forms.CharField(max_length=30, required=False)
```

Comment:
I think you should drop all the `REALMS_HAVE_SUBDOMAINS` code from this PR (but save the code for future reference, since adding that feature would be a great next project!).  That feature will have a good amount of complexity in its own right, and I think it's going to be a better experience to try to get the basic realm creation feature done well and then add that as a future PR.